### DISCUSSION TITLE
Configure as ephemeral runners

### DISCUSSION THREAD

**Discussion by scottdickerson** [machulav/ec2-github-runner#118] [Q&A]
Is there a way to have these EC2 runners configured as ephemeral runners?  Sometimes our stop jobs fail to execute and it leaves the runners sticking around.  I'd like for them to be automatically cleaned up according to this:

https://github.blog/changelog/2022-08-03-github-actions-remove-offline-self-hosted-runners/

**Comment by machulav** [comment]
> Sometimes our stop jobs fail to execute and it leaves the runners sticking around. I'd like for them to be automatically cleaned up

In which place you want to clean up the runners? On GitHub or on AWS?

The stop job does two things: 
- first stop EC2 runner in AWS
- and then remove the related runner from GitHub

If the stop job fails then you need to clean up not only GitHub but also AWS. And the [ephemeral runners](https://github.blog/changelog/2022-08-03-github-actions-remove-offline-self-hosted-runners/) won't help with AWS clean up I suppose.

**Reply by scottdickerson** [reply]
For some reason our EC2 are always cleaned up. It's the GitHub runner that is not. It stays around forever in Offline state. It seems from the logs that a GitHub runner matching that label cannot be found. However I do see a matching GitHub runner in the UI
